### PR TITLE
[WIP] New DynamoDB interface

### PR DESCRIPTION
**Pull request by JoshBurke**
Resolves issue #28 

This PR adds a new dynamo_db.py file in the utils/dataset_db folder. This is an interface for the new dynamoDB instance we have to store our data. several of the collections have already been migrated, but some others are taking time. This migration has not been lossless either, some data can't be moved.
It will likely be changed / remigrated so this PR will be up for a little while.

### CODE REVIEW

File: src/utils/dataset_db/dynamo_db.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+def put_many(table, item_list):
+    """
+    Method to put a list of JSON items in a table.
+    Input:
+        table: the table object you wish to add to
+        item_list: a list of JSON structs containing the items' fields
+    Output:
+        item_count: number of items uploaded successfully
```
Comment (arynh) [inline src/utils/dataset_db/dynamo_db.py:108]: Would be possible or useful to return the metadata that we get from `table.put_item()`?

File: src/utils/dataset_db/dynamo_db.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    """
+    item_count = 0
+    for item in item_list:
+        try:
+            put_item(table, item)
+        except Exception as e:
```
Comment (arynh) [inline src/utils/dataset_db/dynamo_db.py:114]: `except Exception as e` hmmm is there a better way

File: src/utils/dataset_db/dynamo_db.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+import boto3
+import dotenv
+import os
+
+
+db_client = None
+tables_by_name = {}
```
Comment (arynh) [inline src/utils/dataset_db/dynamo_db.py]: What are these global variables for?

Comment (arynh) [reply src/utils/dataset_db/dynamo_db.py]: If they are global, we might want to make them all caps

**Review by arynh — CHANGES_REQUESTED** [review]

**Review by arynh — APPROVED** [review]

File: src/utils/dataset_db/dynamo_db.py (hunk 2 above)
Comment (JoshBurke) [reply src/utils/dataset_db/dynamo_db.py:114]: no!

File: src/utils/dataset_db/dynamo_db.py (hunk 3 above)
Comment (JoshBurke) [reply src/utils/dataset_db/dynamo_db.py]: done

**Pull request merged**